src/main/java/io/supertokens/oauth/OAuth.java

```diff
@@ -90,6 +102,132 @@ public static OAuthAuthResponse getAuthorizationUrl(Main main, AppIdentifier app
+                try {
+                    throw createCustomExceptionFromHttpResponseException(e, OAuthClientRegisterException.class);
+                } catch (NoSuchMethodException | InvocationTargetException | InstantiationException |
+                         IllegalAccessException ex) {
+                    throw new RuntimeException(ex);
+                }
+            } catch (OAuth2ClientAlreadyExistsForAppException e) {
+                //in theory, this is unreachable. We are registering new clients here, so this should not happen.
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public static JsonObject loadOAuthClient(Main main, AppIdentifier appIdentifier, Storage storage, String clientId)
+            throws TenantOrAppNotFoundException, OAuthClientException, InvalidConfigException, StorageQueryException,
+            IOException {
+        OAuthStorage oauthStorage = StorageUtils.getOAuthStorage(storage);
+
+        String adminOAuthProviderServiceUrl = Config.getConfig(appIdentifier.getAsPublicTenantIdentifier(), main).getOAuthProviderAdminServiceUrl();
+
+        if (!oauthStorage.doesClientIdExistForThisApp(appIdentifier, clientId)) {
+            throw new OAuthClientException("Unable to locate the resource", "");
+        } else {
+            try {
+                JsonObject hydraResponse = HttpRequest.sendGETRequest(main, "", adminOAuthProviderServiceUrl + HYDRA_CLIENTS_ENDPOINT + "/" + clientId, null, 10000, 10000, null);
+                return  formatResponseForSDK(hydraResponse);
+            } catch (HttpResponseException e) {
+                try {
+                    throw createCustomExceptionFromHttpResponseException(e, OAuthClientException.class);
+                } catch (NoSuchMethodException | InvocationTargetException | InstantiationException |
+                         IllegalAccessException ex) {
+                    throw new RuntimeException("Something went really wrong!");
+                }
+            }
+        }
+    }
+
+    public static void deleteOAuthClient(Main main, AppIdentifier appIdentifier, Storage storage, String clientId)
+            throws TenantOrAppNotFoundException, OAuthClientException, InvalidConfigException, StorageQueryException,
+            IOException {
+        OAuthStorage oauthStorage = StorageUtils.getOAuthStorage(storage);
+
+        String adminOAuthProviderServiceUrl = Config.getConfig(appIdentifier.getAsPublicTenantIdentifier(), main).getOAuthProviderAdminServiceUrl();
+
+        if (!oauthStorage.doesClientIdExistForThisApp(appIdentifier, clientId)) {
+            throw new OAuthClientException("Unable to locate the resource", "");
+        } else {
+            try {
+                HttpRequest.sendJsonDELETERequest(main, "", adminOAuthProviderServiceUrl + HYDRA_CLIENTS_ENDPOINT + "/" + clientId, null, 10000, 10000, null);
+
+                oauthStorage.removeAppClientAssociation(appIdentifier, clientId);
```

Comment:
this needs to happen the other way around, cause if it succeeds in hydra, but fails during supertokens db, it will lead to an inconsistent state. But if it's the other way around, then failing in hydra, and succeeding in supertokens core is ok (its the same case as in when creating, it succeeds in creating in hydra, but fails in adding to supertokens)

Reply:
Okay. The other way around also leads to an inconsistent state (but at least we treat the client as non existent)

Reply:
done